samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaConsumerProxy.java

```diff
@@ -204,6 +204,7 @@ private Runnable createProxyThreadRunnable() {
         // KafkaSystemConsumer uses the failureCause to propagate the throwable to the container
         failureCause = throwable;
         isRunning = false;
+        ((KafkaSystemConsumer) this.kafkaConsumer).setProxyFailureCause(this.failureCause);
```

Comment:
Should be set before isRunning = false?

Reply:
Does this work? Isn't kafkaConsumer the actual Kafka Consumer, not the SystemConsumer?

Reply:
it doesnt. need some way of giving KafkaConsumer proxy a handle to the SystemConsumer.